### PR TITLE
added wallet claim endpoint

### PR DESCRIPTION
**Pull request by maikelmclauflin**

### CODE REVIEW

File: bat-utils/lib/hapi-server.js [hunk 1]
```diff
@@ -76,6 +76,12 @@ const Server = async (options, runtime) => {
     })
   }
 
+  server.route({
```
Comment (evq) [inline bat-utils/lib/hapi-server.js]: unrelated?

Comment (maikelmclauflin) [reply bat-utils/lib/hapi-server.js]: yes, must have forgotten to take out. removed

File: ledger/controllers/wallet.js [hunk 2]
```diff
@@ -767,6 +797,74 @@ module.exports.initialize = async (debug, runtime) => {
...
+      }, {
+        $push: {
+          paymentIds: paymentId
+        }
+      })
+      // If association worked, perform reverse association, memberId with wallet
+      if (!member) {
+        return reply(boom.conflict())
+      }
+      await wallets.update({ paymentId }, { $set: { memberId } })
```
Comment (evq) [inline ledger/controllers/wallet.js]: @maikelmclauflin can we also set a property like `userCardId` which is the destination card here?

File: ledger/controllers/wallet.js [hunk 3]
```diff
@@ -767,6 +802,86 @@ module.exports.initialize = async (debug, runtime) => {
...
+      })
+      // If association worked, perform reverse association, memberId with wallet
+      if (!member) {
+        return reply(boom.conflict())
+      }
+      await wallets.update({
+        paymentId
+      }, {
+        $set: {
+          userCardId,
```
Comment (evq) [inline ledger/controllers/wallet.js]: we need to drop storing `userCardId` for now

File: test/e2e.integration.test.js [hunk 4]
```diff
@@ -502,6 +503,75 @@ test('ledger : user + grant contribution workflow with uphold BAT wallet', async
...
+
+  await claimCard(anonCardInfo1)
+  await claimCard(anonCardInfo2)
+  await claimCard(anonCardInfo3)
+  await claimCard(anonCardInfo4, 409)
+
+  // redundant calls are fine
+  await claimCard(anonCardInfo1)
+  // // claiming someone elses claimed card results in error
+  // await claimCard(anonCardInfo1, 409)
```
Comment (evq) [inline test/e2e.integration.test.js]: why is this commented out?

**Comment by evq** [comment]
per our offline conversation, recommend adding two e2e sanity tests:
1. you can't submit a transaction to the PUT v2/wallet/{paymentId} that isn't for the settlement test
2. that your balance goes down when you submit a transaction using funds on the anon card

File: ledger/controllers/wallet.js [hunk 5]
```diff
@@ -767,6 +802,86 @@ module.exports.initialize = async (debug, runtime) => {
...
+    const members = runtime.database.get('members', debug)
+
+    const wallet = await wallets.findOne({ paymentId })
+    if (!wallet) {
+      return reply(boom.notFound())
+    }
+
+    const txn = runtime.wallet.validateTxSignature(wallet, signedTx, {
+      destinationValidator: braveJoi.string().guid(),
+      minimum: 0.001
```
[the remark this answers is not in the document]
Comment (maikelmclauflin) [reply ledger/controllers/wallet.js]: minimum set to 0 (to override default of 1 inside of fn)

File: ledger/controllers/wallet.js (hunk 3 above)
Comment (maikelmclauflin) [reply ledger/controllers/wallet.js]: stopped storing `userCardId`
